pkg/distributor/distributor.go

```diff
@@ -1121,6 +1131,11 @@ func (d *Distributor) push(ctx context.Context, pushReq *push.Request) (*mimirpb
 	// so set this flag false and pass cleanup() to DoBatch.
 	cleanupInDefer = false
 
+	if d.cfg.WriteRequestsBufferPoolingEnabled {
+		slabPool := pool2.NewFastReleasingSlabPool[byte](&d.writeRequestBytePool, writeRequestSlabPoolSize)
```

Comment:
I mean, I can see fast releasing slab is an optimization because slabs _may_ be released sooner, but can you confirm it's not strictly required and the base one could be used as well? I just want to make sure I understand rationale.

Reply:
> I can see fast releasing slab is an optimization because slabs may be released sooner, but can you confirm it's not strictly required and the base one could be used as well? I just want to make sure I understand rationale.

Your understanding is correct. It's an optimization to release slabs sooner.

I think this may be actually interesting optimization in environments where average size of write request to ingester is high, and we're also sending requests to many ingesters at once.